Consolidates Provider searches

wgehring-ml opened the pull request:
- Refactors provider search.sjs to be more deterministic & include last
  updated
- Consolidates multiple search methods in PractitionerResourceProvider
  into single search method with optional parameters

wgehring-ml committed:
Consolidates Provider searches

- Refactors provider search.sjs to be more deterministic & include last
  updated
- Consolidates multiple search methods in PractitionerResourceProvider
  into single search method with optional parameters

theflyinggriffin commented on fhir-mapping/src/main/ml-modules/root/data-services/utils/ctsQueryUtils.sjs:
```diff
@@ -0,0 +1,25 @@
+        return fieldToQueryMap[field](values, modifier);
+      }
+
+      const searchValues = egress.searchValuesWithModifier(values, modifier);
```
The application of the modifiers should only happen to string parameters. this transform should probably happen in the field specific functions for the ones that are relevant.

theflyinggriffin commented on fhir-mapping/src/main/ml-modules/root/data-services/practitioner/search.sjs:
```diff
@@ -39,46 +28,63 @@ const codingSystemMap = new Map([
+  },
+  // FUTURE: expand to allow additional timestamp comparison operations?
+  _lastUpdated(values, modifier) {
+    return cts.jsonPropertyRangeQuery('ingestTimestamp', '>=', xs.dateTime(`${values[0]}T00:00:00`));
```
This ignores the modifier that the user passes in with the date. While this is a perfect lift of the DS that it is replacing, it is incorrect and should use the modifiers that are provided. {<, <=, =, >=, >} should be supported, for an example of how to do dateTime equality checks with only a supplied date see the location search data service.

theflyinggriffin commented on hapi-server/src/main/java/com/example/fhirexample/ResourceProvider/PractitionerResourceProvider.java:
```diff
@@ -184,85 +89,44 @@ public Practitioner read(@IdParam IdType theId) {
+        @OptionalParam(name = Practitioner.SP_FAMILY) StringAndListParam family,
+        @OptionalParam(name = Practitioner.SP_NAME) StringAndListParam name,
+        @OptionalParam(name = Practitioner.SP_GIVEN) StringAndListParam given,
+        @OptionalParam(name = Constants.PARAM_LASTUPDATED) DateRangeParam lastUpdated,
```
Instead of using a DateRange param, use the DateParam. This will allow {<, <=, =, >=, >} options for the modifiers and will send those modifiers to the DS. we could also update this to be DateAndListParam to allow more options for the query (i.e. more than 1 criteria for the date parameter allowing date ranges in addition to the other options)

wgehring-ml committed:
Addresses comments on PR

- Fixes/makes ctsQueryUtils text searches more flexible
- Allows more than >= modifier on practitioner lastUpdated search
- Changes practitioner HAPI lastUpdated parameter to a DateAndListParam
- Uses dateModifierMap to translate DateAndListParam/DateOrListParam
  modifiers to a format recognized by the .sjs files
